crates/supervisor/core/src/l1_watcher/watcher.rs

```diff
@@ -88,13 +88,13 @@ where
                 }
                 latest_block = latest_head_stream.next() => {
                     if let Some(latest_block) = latest_block {
-                        info!(target: "supervisor::l1_watcher", "Latest L1 block received: {:?}", latest_block.header.number);
+                        trace!(target: "supervisor::l1_watcher", "Latest L1 block received: {:?}", latest_block.header.number);
                         previous_latest_block = self.handle_new_latest_block(latest_block, previous_latest_block).await;
                     }
                 }
                 finalized_block = finalized_head_stream.next() => {
                     if let Some(finalized_block) = finalized_block {
-                        info!(target: "supervisor::l1_watcher", "Finalized L1 block received: {:?}", finalized_block.header.number);
+                        trace!(target: "supervisor::l1_watcher", "Finalized L1 block received: {:?}", finalized_block.header.number);
```

Comment:
instead of logging here, Can we log in the handle_new_latest_block method?
Also please log the previous and incoming block